core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache.evictor;
+
+import alluxio.client.file.cache.CacheEvictor;
+import alluxio.client.file.cache.PageId;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * LFU client-side cache eviction policy.
+ */
+@ThreadSafe
+public class LFUCacheEvictor implements CacheEvictor {
+  private static final int PAGE_MAP_INIT_CAPACITY = 200;
+  private static final float PAGE_MAP_INIT_LOAD_FACTOR = 0.75f;
+  private static final int BUCKET_MAP_INIT_CAPACITY = 32;
+  private static final float BUCKET_MAP_INIT_LOAD_FACTOR = 0.75f;
+
+  private final Map<PageId, Integer> mPageMap = new HashMap<>(
+      PAGE_MAP_INIT_CAPACITY, PAGE_MAP_INIT_LOAD_FACTOR);
+
+  private final Map<Integer, Set<PageId>> mBucketMap =
+      new HashMap<>(BUCKET_MAP_INIT_CAPACITY, BUCKET_MAP_INIT_LOAD_FACTOR);
+  private int mMinBucket = -1;
+  private final double mLogBase;
+
+  /**
+   * Required constructor.
+   *
+   * @param conf Alluxio configuration
+   */
+  public LFUCacheEvictor(AlluxioConfiguration conf) {
+    mLogBase = conf.getDouble(PropertyKey.USER_CLIENT_CACHE_EVICTOR_LFU_LOGBASE);
+  }
+
+  private int getBucket(int count) {
+    return (int) (Math.log(count) / Math.log(mLogBase));
+  }
+
+  private void addPageToBucket(PageId pageId, int bucket) {
+    mBucketMap.compute(bucket, (bucketKey, pageSet) -> {
+      Set<PageId> set = pageSet == null ? new LinkedHashSet<>() : pageSet;
```

Comment:
If we change `LinkedHashSet` to `LinkedHashMap` with access order (check `LRUCacheEvictor`), don't we have a LRU bucket in that case? @calvinjia

Reply:
Done.